### ISSUE TITLE
Dio Smart Retry 6.0.0 not working; version 5.0.0 works fine

### ISSUE THREAD

**Issue by rafamizes** [rodion-m/dio_smart_retry#41]
I updated `dio_smart_retry` to version `6.0.0` and retries stopped working, so I reverted back to version `5.0.0` — which works just fine.

By the way, I'm also using `MultipartFileRecreatable` due to multipart uploads.

See below how we are instantiating `RetryInterceptor` objects:

```dart
    List<Duration> retryDelays = const [
      Duration(seconds: 1), // waits 1s before the first retry.
      Duration(seconds: 2), // waits 2s before the second retry.
      Duration(seconds: 4), // waits 4s before the third retry.
      Duration(seconds: 8), // waits 8s before the fourth — and last — retry.
    ],
    final dio = Dio(BaseOptions(baseUrl: Env.url()));
    dio.interceptors.add(
      RetryInterceptor(
        dio: dio,
        logPrint: logger.warn,
        retries: retryDelays.length,
        retryDelays: retryDelays,
     ),
   );
```

ps: the dio version in pubspec.yaml is `5.6.0`.

**Comment by rodion-m** [comment]
I think it was solved by #42. Fill free to reopen it if it's not.

**Comment by rafamizes** [comment]
Ok, I will check it out.
